### PR TITLE
feat(dataZoom): `valueRange` for `dataZoom` event

### LINKED ISSUE #10700: add startValue and endValue parameters in zoom event triggered by slider
### What problem does this feature solve?
When user uses slider to zoom the chart, the event triggered does not have parameters `startValue` and `endValue`. It only has `start` and `end` percentage values. 

### Version
4.2.1
 
### Reproduction link
https://www.echartsjs.com/examples/en/editor.html?c=line-simple
 
### Steps to reproduce
- Please copy the following code into input on the left side.

 
```javascript
option = {
    title: {
        text: 'ECharts 入门示例'
    },
    tooltip: {},
    legend: {
        data:['销量']
    },
    xAxis: {
        data: ["衬衫","羊毛衫","雪纺衫","裤子","高跟鞋","袜子"]
    },
    yAxis: {},
    series: [{
        name: '销量',
        type: 'line',
        data: [5, 20, 36, 10, 10, 20]
    }],
    dataZoom: [
        {
            type: 'slider',
            show: true,
            realtime: false
        },    
    ]
};

myChart.on('dataZoom', function (params) {
    console.log(params);
    
    // should return exact values, but returns undefined
    console.log(params.startValue);
    console.log(params.endValue);
});
```
 
- Then try to zoom the chart and have a look at console.
 
### What is expected?
As described in the [docs](https://echarts.apache.org/en/api.html#events.datazoom), datazoom event includes `startValue` and `endValue` in the payload if the event is triggered by using toolbox. But I think the datazoom event (triggered by the slider) should also take parameters `startValue` and `endValue` besides `start` and `end` percentage parameters.
 
### What is actually happening?
`startValue` and `endValue` parameters are not included in the payload of the event.

Thank you.

### PR DESCRIPTION
**Pull request by susiwen8**
## Brief Information

This pull request is in the type of:

- [ ] bug fixing
- [x] new feature
- [ ] others

### What does this PR do?

Expose `startValue` and `endValue` at `dataZoom` event

### Fixed issues

- Close #10700 

## Document Info

One of the following should be checked.

- [ ] This PR doesn't relate to document changes
- [ ] The document should be updated later
- [x] The document changes have been made in apache/echarts-doc#10700

### ZRender Changes

- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

N.A.

### Merging options

- [x] Please squash the commits into a single one when merging.

### CODE REVIEW

**Review by Ovilia — APPROVED** [review]

**Comment by Ovilia** [comment]
The document should be updated by removing "only exists in zoom event of triggered by toolbar".

File: src/component/dataZoom/SliderZoomView.ts [hunk 1]
```diff
@@ -816,6 +817,7 @@ class SliderZoomView extends DataZoomView {
                         start: range[0], end: range[1]
                     }).valueWindow
                     : axisProxy.getDataValueWindow();
+                this._valueRange = dataInterval;
```
Comment (Ovilia) [inline src/component/dataZoom/SliderZoomView.ts:820]: So if `showDetail` is false, this value range is not updated. Is this expected?

Comment (susiwen8) [reply src/component/dataZoom/SliderZoomView.ts:820]: No, I will rearrage it

**Comment by susiwen8** [comment]
> The document should be updated by removing "only exists in zoom event of triggered by toolbar".

Created a PR for it.

**Review by Ovilia — COMMENTED** [review]
LGTM but I would like @pissang and @100pah to confirm that the `if (dataZoomModel.get('showDetail'))` condition was added by mistake.

**Comment by pissang** [comment]
Looks good. `if (dataZoomModel.get('showDetail'))` is not a mistake. It's just the old logic only used `dataInterval` to display the labels and didn't save it to`_dataRange`.

**Review by pissang — DISMISSED** [review]

**Comment by pissang** [comment]
Seems the valueRange is updated after the action is dispatched. I guess in the event `startValue` and `endValue` will not be the latest value?

**Comment by SevenOutman** [comment]
Any update on this PR? This would be really helpful as there's merely no way to map `start`/`end` percentage to values in the same way as ECharts does.

**Comment by susiwen8** [comment]
> Seems the valueRange is updated after the action being dispatched. I guess in the event `startValue` and `endValue` will not be the latest value? A test will be good to verify it

@pissang @Ovilia @plainheart  The missing person has returned, Sorry I haven't checked this in time, had very busy two months.

The timing for calculating `_valueRange` has moved after the calculation of `range`. This might be the better way, cause `dataZoom` event rely on `range` as well as `_valueRange`, so this two variable could be calculated at same time

**Comment by ddubrava** [comment]
By the way, if the xAxis.type is set to time, the dataZoom.startValue and dataZoom.endValue properties will return formatted values, which may make it difficult to map them back to the original data. For example, if your dataset value is 1682850281010, the dataZoom.startValue will be 1682850281399.8394. Even if you use an ISO string format instead of a timestamp, the dataZoom object still returns a formatted value

**Pull request closed without merging**